sdk/core/Azure.Core/README.md

```diff
@@ -40,110 +40,156 @@ These options are passed as a parameter that extends ```ClientOptions``` class e
+        Mode = RetryMode.Fixed
+    },
+    Diagnostics =
+    {
+        IsLoggingContentEnabled = true,
+        ApplicationId = "myApplicationId"
+    }
+};
+
+SecretClient client = new SecretClient(new Uri("http://example.com"), new DefaultAzureCredential(), options);
 ```
 
+More on client configuration in [client configuration samples](samples/Configuration.md)
+
 ### Accessing HTTP Response Details Using ```Response<T>```
 _Service clients_ have methods that can be used to call Azure services. 
 We refer to these client methods _service methods_.
 _Service methods_ return a shared Azure.Core type ```Response<T>``` (in rare cases its non-generic sibling, a raw ```Response```).
 This type provides access to both the deserialized result of the service call, 
 and to the details of the HTTP response returned from the server.
 
-```csharp
-public async Task UsingResponseOfT()
+```C# Snippet:ResponseTHelloWorld
+// create a client
+var client = new SecretClient(new Uri("http://example.com"), new DefaultAzureCredential());
+
+// call a service method, which returns Response<T>
+Response<KeyVaultSecret> response = await client.GetSecretAsync("SecretName");
+
+// Response<T> has two main accessors.
+// Value property for accessing the deserialized result of the call
+KeyVaultSecret secret = response.Value;
+
+// .. and GetRawResponse method for accessing all the details of the HTTP response
+Response http = response.GetRawResponse();
+
+// for example, you can access HTTP status
+int status = http.Status;
+
+// or the headers
+foreach (HttpHeader header in http.Headers)
 {
-    // create a client
-    var client = new BlobContainerClient(connectionString, "container");
+    Console.WriteLine($"{header.Name} {header.Value}");
+}
+```
+
+More on response types in [response samples](samples/Response.md)
+
+### Setting up console logging
 
-    // call a service method, which returns Response<T>
-    Response<ContainerItem> response = await client.GetPropertiesAsync();
+To create an Azure SDK log listener that outputs messages to console use `AzureEventSourceListener.CreateConsoleLogger` method.
 
-    // Response<T> has two main accessors. 
-    // Value property for accessing the deserialized result of the call
-    ContainerItem container = response.Value;
+```C# Snippet:ConsoleLogging
+// Setup a listener to monitor logged events.
+using AzureEventSourceListener listener = AzureEventSourceListener.CreateConsoleLogger();
+```
 
-    // .. and GetRawResponse method for accessing all the details of the HTTP response
-    Response http = response.GetRawResponse();
+More on logging in [configuration samples](samples/Configuration.md)
 
-    // for example, you can access HTTP status
-    int status = http.Status;
+### Reporting Errors ```RequestFailedException```
 
-    // or the headers
-    foreach(HttpHeader header = http.Headers) {
-        Console.WriteLine($"{header.Name} {header.Value}");
-    }
+```C# Snippet:RequestFailedException
+try
+{
```

Comment:
"require" (missing an "i")

Reply:
Also, the sentence reads odd starting around "their status such operations". Probably should rephrase.